presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -897,9 +900,29 @@ else if (executionException instanceof PrestoSparkExecutionException) {
                 PlanFragment childFragment = child.getFragment();
                 if (childFragment.getPartitioningScheme().getPartitioning().getHandle().equals(FIXED_BROADCAST_DISTRIBUTION)) {
                     RddAndMore<PrestoSparkSerializedPage> childRdd = createRdd(child, PrestoSparkSerializedPage.class);
+
+                    // TODO: The driver might still OOM on a very large broadcast, think of how to prevent that from happening
                     List<PrestoSparkSerializedPage> broadcastPages = childRdd.collectAndDestroyDependencies().stream()
                             .map(Tuple2::_2)
                             .collect(toList());
+
+                    int compressedBroadcastSizeInBytes = broadcastPages.stream()
+                            .mapToInt(page -> page.getBytes().length)
+                            .sum();
+                    int uncompressedBroadcastSizeInBytes = broadcastPages.stream()
+                            .mapToInt(PrestoSparkSerializedPage::getUncompressedSizeInBytes)
+                            .sum();
+                    DataSize maxBroadcastSize = getQueryMaxBroadcastMemory(session);
+                    long maxBroadcastSizeInBytes = maxBroadcastSize.toBytes();
```

Comment:
just double check: this is consistent with how Presto classic checks max broadcast size right?

Also I feel in theory we can do the check on driver ? (before broadcast the data)

Reply:
> just double check: this is consistent with how Presto classic checks max broadcast size right?

Yeah. The only difference that Presto Classic enforces that limit on a consumer side. Here we enforce it on a driver before even starting the downstream stage that is supposed to consume the broadcasted table.